src/flightRecorder.cpp

```diff
@@ -998,9 +1000,16 @@ class Recording {
         writeStackTraces(buf, &lookup);
         writeMethods(buf, &lookup);
         writeClasses(buf, &lookup);
-        writePackages(buf, &lookup);
+
+        if(!tryWritePackages(buf, &lookup)) {
+            poolCount--;
+        }
+
         writeSymbols(buf, &lookup);
         writeLogLevels(buf);
+
+        // Patch pool count.
+        buf->put8(poolCountOffset, poolCount);
```

Comment:
It's not that trivial.
Buffer can be flushed while dumping other constant pools.

Reply:
Also, the problem is relevant to other dictionaries, too - for example, when a recording contains no samples.
An alternative workaround that I thought of is to populate dictionary with a dummy element if it is empty.